pyteal/ast/abi/array_dynamic.py

```diff
@@ -100,11 +100,34 @@ def length(self) -> Expr:
 DynamicArray.__module__ = "pyteal.abi"
 
 
+class DynamicBytesTypeSpec(DynamicArrayTypeSpec):
```

Comment:
nit: 
```suggestion
class DynamicBytesTypeSpec(DynamicArrayTypeSpec[ByteType]):
```

Reply:
+1 to this, except `ByteType` should just be `Byte`